cli/tests/integration/device_option.rs

```diff
@@ -93,3 +93,148 @@ async fn missing_option(tmp_path: TmpPath) {
             .and(predicates::str::contains(&available_devices_string_list)),
     );
 }
+
+#[rstest::rstest]
+#[tokio::test]
+async fn multiple_device_found(tmp_path: TmpPath) {
+    let config = super::get_testenv_config();
+    let url = super::new_environment(&tmp_path, None, config, false)
+        .await
+        .unwrap()
+        .unwrap();
+
+    let org_id = super::unique_org_id();
+    let org_addr = crate::commands::create_organization::create_organization_req(
+        &org_id,
+        &url,
+        crate::testenv_utils::DEFAULT_ADMINISTRATION_TOKEN,
+    )
+    .await
+    .unwrap();
+
+    let client_config = libparsec::ClientConfig::default();
+    let first_available_device =
+        crate::commands::bootstrap_organization::bootstrap_organization_req(
+            client_config.clone(),
+            org_addr,
+            "dev1".parse().unwrap(),
+            libparsec::HumanHandle::new("first@dev1.com", "First").unwrap(),
+            crate::testenv_utils::DEFAULT_DEVICE_PASSWORD
+                .to_string()
+                .into(),
+        )
+        .await
+        .unwrap();
+
+    let first_device = libparsec::load_device(
+        &client_config.config_dir,
+        &libparsec::DeviceAccessStrategy::Password {
+            key_file: libparsec::get_default_key_file(
+                &client_config.config_dir,
+                &first_available_device.device_id,
+            ),
+            password: crate::testenv_utils::DEFAULT_DEVICE_PASSWORD
+                .to_string()
+                .into(),
+        },
+    )
+    .await
+    .unwrap();
+
+    let first_auth_cmds = libparsec::AuthenticatedCmds::new(
+        &client_config.config_dir,
+        first_device.clone(),
+        libparsec::ProxyConfig::default(),
+    )
+    .unwrap();
+
+    let first_dev_id_prefix = &first_device.device_id.hex()[..8];
+
+    let second_device = create_second_device(
+        &first_auth_cmds,
+        &first_device,
+        client_config,
+        first_dev_id_prefix,
+    )
+    .await
+    .unwrap();
+
+    let mut devices = [first_device, second_device];
+    devices.sort_by_cached_key(|d| d.device_id.hex());
+
+    let mut available_devices_string_list = String::new();
+    format_device::<3>(&devices, &mut available_devices_string_list).unwrap();
+
+    crate::assert_cmd_failure!("list-users", "--device", first_dev_id_prefix).stderr(
+        predicates::str::contains(format!(
+            "Error: Multiple devices found for `{first_dev_id_prefix}`:"
```

Comment:
Note having two devices begining with the same characters is possible and shouldn't be an issue.

In Parsec v2 we would start with a 3 characters short hex id, then make it bigger whenever the number of characters become too small to uniquely identify each device.

So I think we should open an issue about this

Reply:
I've openned an issue: #8034